packages/uploadthing/src/internal/handler.ts

```diff
@@ -63,42 +63,59 @@ const createUTFetch = (
   };
 };
 
-const fileCountLimitHit = (
+const fileCountBoundsCheck = (
   files: { name: string }[],
   routeConfig: ExpandedRouteConfig,
 ) => {
   const counts: Record<string, number> = {};
 
   files.forEach((file) => {
     const type = getTypeFromFileName(file.name, objectKeys(routeConfig));
-
-    if (!counts[type]) {
-      counts[type] = 1;
-    } else {
-      counts[type] += 1;
-    }
+    counts[type] = (counts[type] || 0) + 1;
   });
 
   for (const _key in counts) {
     const key = _key as FileRouterInputKey;
     const count = counts[key];
-    const limit = routeConfig[key]?.maxFileCount;
+    const min = routeConfig[key]?.minFileCount;
+    const max = routeConfig[key]?.maxFileCount;
 
-    if (!limit) {
+    if (typeof min !== "number" || typeof max !== "number") {
       logger.error(routeConfig, key);
+      const configKey =
+        typeof min !== "number" ? "minFileCount" : "maxFileCount";
       throw new UploadThingError({
         code: "BAD_REQUEST",
-        message: "Invalid config during file count",
-        cause: `Expected route config to have a maxFileCount for key ${key} but none was found.`,
+        message: `Invalid config during file count - missing ${configKey}`,
+        cause: `Expected route config to have a ${configKey} for key ${key} but none was found.`,
       });
```

Comment:
i think this is actually the wrong error message?

the `minFileCount`/`maxFileCount` can never be undefined since we backfill defaults, right?

it's the `routeConfig[key]` which may be nullable. So the error message should probably be related to that?